aws-codeartifact-domain/src/test/java/software/amazon/codeartifact/domain/CreateHandlerTest.java

```diff
@@ -136,6 +136,48 @@ public void handleRequest_simpleSuccess() {
 
     }
 
+    @Test
+    public void handleRequest_simpleSuccess_retried() {
```

Comment:
This test makes me think that we can have a create request that will succeed on the first attempt, even if the resource exists. which we def do not want. We dont want people to be able to do resource adoption by just having a resource exist.

Reply:
If the resource exists, the creation will still fail with a 409. This test is testing if the handler will succeed if the resource exists **and** callbackContext.isCreated() == true. This value can only be set during the lambda's invokation so 

In this test, I am setting the isCreated Flag [here](https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-codeartifact/pull/57/files#diff-1fb744651090160b8ba2005b1e30053f8b09de6262c121f86182c7bc50831d73R163) to mimic this situation like

```
        CallbackContext context = new CallbackContext();
        context.setCreated(true);
        final ProgressEvent<ResourceModel, CallbackContext> response = handler.handleRequest(proxy, request, context, proxyClient, logger);
```


This is the handler log if the domain exists and we call this create handler

```
[CREATE] invoking handler...
isCreated flag false
Failed to execute remote function: {null (Service: Codeartifact, Status Code: 409, Request ID: 2f096ab6-f95e-4169-82b1-f30ebbd38074, Extended Request ID: null)}
Resource of type 'AWS::CodeArtifact::Domain' with identifier 'domain-retried' already exists. in a CREATE action on a null: software.amazon.cloudformation.exceptions.CfnAlreadyExistsException: Resource of type 'AWS::CodeArtifact::Domain' with identifier 'domain-retried' already exists.
```